portfolio/src/main/webapp/Projects/snakeProject/sketch.js

```diff
@@ -0,0 +1,62 @@
+var s;
+var scl = 20;
+var food;
+var up = true;
+var down = true;
+var left = true;
+var right = true;
+function setup() {
+  createCanvas(600, 600);
+  s = new Snake();
+  foodLocation();
+}
+function foodLocation() {
+  var columns = floor(width / scl);
+  var rows = floor(height / scl);
+  food = createVector(floor(random(columns)), floor(random(rows)));
+  food.mult(scl);
+}
+
+function draw() {
+  background(51);
+  s.destroy();
+  s.show();
+  s.move();
+  if (s.eat(food)) {
+    foodLocation();
+  }
+  frameRate(10);
+  fill(153, 42, 99);
+  rect(food.x, food.y, scl, scl);
+}
+function keyPressed() {
+  if (keyCode === UP_ARROW && up) {
+    s.direction(0, -1);
```

Comment:
Optional: I know this isn't strictly part of the SPS project but you could consider using something [enum-like](https://www.sohamkamani.com/blog/2017/08/21/enums-in-javascript/) here.

I could see this code being transformed from:
```javascript
var up = true;
var down = true;
var left = true;
var right = true;
...
if (keyCode == UP_ARROW && up) {
  s.direction(0, -1);
  up = true;
  down = false;
  left = true;
  right = true;
} ...
```
to something like:
```javascript
const directions = {
    UP: 'up',
    DOWN: 'down',
    LEFT: 'left',
    RIGHT: 'right'
}
var dir = null;
...
if (keyCode == UP_ARROW && dir != directions.DOWN) {
  dir = directions.UP;
  s.direction(dir);
} ...
```

Again, this is completely optional. It's a cool project regardless!